### ISSUE TITLE
cordova ios 6.1.0 wkwebview all xhr POST requests failing and are converting as blob request.

### ISSUE THREAD

**Issue by knkavya** [Raphcal/cordova-plugin-cors#11]
Hi,

I have migrated the Cordova app from UIWebview to WKWebview.I have a native scanner screen that is invoked from the webpage and to invoke this I am using the "cordova-plugin-cors".
Which is working fine by launching a scanner. But on the same page, the other ajax post calls are getting failed and they are converted as blob requests, along with that it gives certificate trust error logs.

I tried removing the "cordova-plugin-cors" plugin post that all the post ajax call works perfectly. But Scanner will not be invoked without xhr plugin.

All the calls are getting invoked using the xhr.js file where the post calls are failing.
Can someone please help me to fix this issue?

<img width="1029" alt="XHR plugin issue" src="https://user-images.githubusercontent.com/15210891/102050286-8b243b00-3e08-11eb-8b07-41b6af8f58fe.png">

**Comment by Raphcal** [comment]
When using this plugin, XHR requests should not be visible from the web browser since XHR requests are sent by the plugin (in native Objective-C code).
Are you sure this is caused by this plugin ? How can I reproduce your problem ?

**Comment by knkavya** [comment]
As informed earlier I have a native scanner screen that is invoked from the server-side and to invoke this I am using the "cordova-plugin-cors" plugin.
But whenever the POST-call enters into xhr-polyfill.js send request my ajax post calls are getting failed and they are converted as blob request.
I have debugged this issue by putting a breakpoint inside the xhr.js file using app inspector where all the server calls for the app are getting invoked from this file itself.
Looks like it is an issue with xhr.js file send request for the ajax post calls.

**Comment by Raphcal** [comment]
If the problem occurs when you send file, can you paste the code you are using to send a file ?

The request is initiated by FileReader.js. Is it a script you wrote or does it come from an other plugin ?

**Comment by knkavya** [comment]
Hi Raphcal

Problem occurs only for the post request.
As the callback enters send method inside RCAXMLHttpRequest file below are my request details.

method : POST
path : Requested URL
headers : {
    Accept = "application/json, text/javascript, */*; q=0.01";
    "Content-Type" = "application/x-www-form-urlencoded; charset=UTF-8";
    "User-Agent" = "Mozilla/5.0 (iPhone; CPU iPhone OS 13_5 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148/RwdApp";
    "X-Requested-With" = XMLHttpRequest;
}
responseType : text

When the request enters the 
NSURLSessionDataTask *task = [_session dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable urlResponse, NSError * _Nullable error) {

Gives me security issue for the request as

Task <AE834EED-BE01-44C1-87BC-728831452C53>.<6> finished with error [-1202] Error Domain=NSURLErrorDomain Code=-1202 "The certificate for this server is invalid. You might be connecting to a server that is pretending to be “store.*****.***.**” which could put your confidential information at risk."

and it is converting the request as a blob request.

**Comment by Raphcal** [comment]
Thank you for the details. I will try to reproduce your problem.

In the meantime, did you try adding a transport security exception to ignore invalid certificates error ?

**Comment by knkavya** [comment]
Yes. I have tried adding transport security exceptions to ignore invalid certificates, but the issue is still the same.

For other requests, I am not getting a security error. Only for the post request, I am getting an error.

**Comment by knkavya** [comment]
Hi Raphcal,

Any update on this? are you able to reproduce my issue.

**Comment by gangishettisahiti** [comment]
Hi Kavya,
 Have you found any solution for the above problem. We are facing the same issue in our application. Kindly suggest any work around for this.
